Sources/MapboxCoreNavigation/TileStore.swift

```diff
@@ -19,42 +19,43 @@ extension TileStore {
      */
     public func containsLatestNavigationTiles(forCacheLocation cacheLocation: TileStoreConfiguration.Location = .default,
                                               completion: @escaping ContainsCompletion) {
-        let descriptors = [TilesetDescriptorFactory.getLatest(forCacheLocation: cacheLocation)]
-        
-        __getAllTileRegions { expected in
-            if expected.isValue(), let regions = expected.value as? Array<TileRegion> {
-                let lock = NSLock()
-                var count = regions.count
-                var result: Bool? = true
+        TilesetDescriptorFactory.getLatest(forCacheLocation: cacheLocation,
+                                           completionQueue: .main) { latestTilesetDescriptor in
```

Comment:
Since now the callback has explicit queue definition we should either document that it will always be called on a main thread or allow customising it as well.

Reply:
Well, actually a thread on which containsLatestNavigationTiles completion will be called stays the same as before. Here we are getting just the tile descriptor on the main thread.